overlord/snapstate/handlers.go

```diff
@@ -1147,9 +1149,43 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 		}
 		return copyDataErr
 	}
+
+	// the migration hasn't been done - do it now
+	if opts.UseHidden && !opts.MigratedToHidden {
+		if err := m.migrateToHiddenDir(t); err != nil {
```

Comment:
Sure, the idea was to separate the migration logic in another function but, in that case, I'd also move the migration call here. Otherwise, we're left with a 2 line function with just some of the migration logic